src/compiler/parser.rs

```diff
@@ -501,9 +514,27 @@ impl ParserType {
         self.next_token_without_check()
     }
 
+    fn re_scan_template_token(&self, is_tagged_template: bool) -> SyntaxKind {
+        self.set_current_token(self.scanner().re_scan_template_token(
+            Some(&|message, length| self.scan_error(message, length)),
```

Comment:
Could expose an eg `.scanner_on_error()` method returning this wrapped closure (that gets injected as the scanner `on_error` callback where needed) to DRY this up if that didn't eg cause lifetime issues